doc/DeveloperGuide.md

```diff
@@ -88,6 +88,30 @@ Use case ends.
 > 3a1. AddressBook shows an error message <br>
   Use case resumes at step 2
 
+#### Use case 2: Rename an existing tag
+
+**MSS**
+1. User requests to list tags
+2. AddressBook shows a list of tags
+3. User requests to rename a specific tag in the list
+4. AddressBook renames the tag<br>
+Use case ends.
+
+**Extensions**
+
+2a. The list is empty
+
+> Use case ends
+
+3a. The given index is invalid
+
+> 3a1. AddressBook shows an error message <br>
+  Use case resumes at step 2
+  
+3b. The given replacement tag name is invalid
```

Comment:
- What happens when the new tag is already present in the address book?
- What happens if the new tag is same as the old tag?